### DISCUSSION TITLE
Tag to bool deserialization

### DISCUSSION THREAD

**Discussion by Symfony2** [YAXLib/YAXLib#240] [Q&A]
`Hello` everyone, I need to deserialize such behavior:
```
<test>
     <gop/> <----- Has tag 'true'
</test>

<test>
      <----- no tag 'false'
</test>

```
class Test{
public bool Gop {get;set;}
}
Any help appreciated

**Comment by axunonb** [comment]
You could implement an [`ICustomSerializer<bool>`](https://github.com/YAXLib/YAXLib/blob/master/YAXLib/ICustomSerializer.cs) for the `Gop` property. You'll find an example for a [field serializer here](https://github.com/YAXLib/YAXLib/blob/master/YAXLibTests/SampleClasses/CustomSerialization/FieldLevelSerializer.cs).

**Reply by Symfony2** [reply]
Thank you for reply. But for
`<test>
      <----- no tag 'false'
</test>
`
For this case you will not reach [ICustomSerializer<bool>](https://github.com/YAXLib/YAXLib/blob/master/YAXLib/ICustomSerializer.cs)
YAXLib.Exceptions.YAXElementMissingException
  HResult=0x80131500
  Message=No elements with this name found: './Gop'.



`
public class NullBoolSerializer : ICustomSerializer<bool>
{
    

    public void SerializeToAttribute(bool objectToSerialize, XAttribute attrToFill, ISerializationContext serializationContext)
    {
        throw new NotImplementedException();
    }

    public void SerializeToElement(bool objectToSerialize, XElement elemToFill, ISerializationContext serializationContext)
    {
        throw new NotImplementedException();
    }

    public string SerializeToValue(bool objectToSerialize, ISerializationContext serializationContext)
    {
        throw new NotImplementedException();
    }

    public bool DeserializeFromAttribute(XAttribute attribute, ISerializationContext serializationContext)
    {
        throw new NotImplementedException();
    }

    public bool DeserializeFromElement(XElement element, ISerializationContext serializationContext)
    {
        return element.IsEmpty;
    }

    public bool DeserializeFromValue(string value, ISerializationContext serializationContext)
    {
        throw new NotImplementedException();
    }
}
`

`
[YAXSerializeAs("Gop")]
    [YAXCustomSerializer(typeof(NullBoolSerializer))]
    [YAXErrorIfMissed(YAXExceptionTypes.Ignore, DefaultValue = false)]
    public bool Gop{ get; set; }`
 
Only above helped to achieve such behevior/ Don't know maybe there is another ellegant way ?

**Reply by axunonb** [reply]
Yes, that's the way to handle your use case with YAXLib.
